packages/pouchdb-core/src/setup.js

```diff
@@ -45,6 +45,8 @@ PouchDB.adapter = function (id, obj, addToPreferredAdapters) {
 PouchDB.plugin = function (obj) {
   if (typeof obj === 'function') { // function style for plugins
     obj(PouchDB);
+  } else if (Object.keys(obj).length === 0){
```

Comment:
good catch - I think that's why my test had not thrown errors before, as I was testing with a string.